src/main/scala/is/hail/asm4s/FunctionBuilder.scala

```diff
@@ -74,87 +157,98 @@ class FunctionBuilder[F >: Null](parameterTypeInfo: Array[MaybeGenericTypeInfo[_
   cn.superName = "java/lang/Object"
   cn.interfaces.asInstanceOf[java.util.List[String]].add("java/io/Serializable")
 
-  def descriptor: String = s"(${ parameterTypeInfo.map(_.base.name).mkString })${ returnTypeInfo.base.name }"
+  var methods: mutable.ArrayBuffer[MethodBuilder] = new mutable.ArrayBuffer[MethodBuilder](16)
 
-  val mn = new MethodNode(ACC_PUBLIC, "apply", descriptor, null, null)
+  val apply = new MethodBuilder(this, "apply", parameterTypeInfo.map(_.base), returnTypeInfo.base)
```

Comment:
this is confusing me a bit. Apply is a somewhat-reserved word in scala. Can it be renamed something a bit more descriptive?

Reply:
apply_method?

Reply:
I think that's clearer, yeah